cookbooks/aws-parallelcluster-slurm/recipes/update/update_compute.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-slurm
+# Recipe:: update_compute
+#
+# Copyright 2024 Amazon.com, Inc. and its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+unless on_docker? || kitchen_test? && !node['interact_with_ddb']
```

Comment:
what about having all this logic `unless on_docker? || kitchen_test? && !node['interact_with_ddb']` into the `save_cluster...` function?

Is responsibility of the function to know if it can be called during kitchen tests or in docker, it's not a responsibility of the caller to know how it's implemented. This will help to avoid having the same guards in every place we're calling it.

Reply:
Very good point! I completely agree